cmdstanpy/compiler_opts.py

```diff
@@ -165,6 +170,13 @@ def validate_stanc_opts(self) -> None:
                     self._cpp_options = {'STAN_OPENCL': 'TRUE'}
                 else:
                     self._cpp_options['STAN_OPENCL'] = 'TRUE'
+            elif key.startswith('O'):
+                if has_O_flag:
+                    raise UserWarning('More than one of (O, O1, O2, Oexperimental)'
+                                      'optimizations passed. Only the last one will'
+                                      'be used')
```

Comment:
We should use `get_logger().warning(...)` here to match the rest of what we do.

Besides that, I think this just needs a test or two and then it should be good. Thanks @grburgess !

Reply:
Cool. I will switch that and add the tests within an hour or so.

Reply:
Should be ready now. I tests adding more than one --O so that the logger gets covered as well.